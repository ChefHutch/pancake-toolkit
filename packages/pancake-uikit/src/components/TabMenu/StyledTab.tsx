import styled from "styled-components";
import { TabMenuItemProps } from "./types";

const StyledTab = styled.div<TabMenuItemProps>`
  margin: 0 2px;
  padding: 8px 22px;
  border-radius: 16px 16px 0 0;
  background-color: ${(props) => (props.isActive ? "red" : "blue")};
`;

export default StyledTab;
