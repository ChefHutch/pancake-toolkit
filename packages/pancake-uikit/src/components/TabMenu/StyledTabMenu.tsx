import styled from "styled-components";

const StyledTabMenu = styled.div``;

export default StyledTabMenu;
