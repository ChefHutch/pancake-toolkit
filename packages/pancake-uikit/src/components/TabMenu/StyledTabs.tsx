import styled from "styled-components";

export const ActiveTab = styled.div``;

export const InactiveTab = styled.div``;
