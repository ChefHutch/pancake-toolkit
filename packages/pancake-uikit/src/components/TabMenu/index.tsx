export { default as TabMenu } from "./TabMenu";
export { default as TabMenuItem } from "./TabMenuItem";
export type { TabMenuProps, TabMenuItemProps } from "./types";
