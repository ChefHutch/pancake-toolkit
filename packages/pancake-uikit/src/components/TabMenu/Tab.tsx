import React from "react";
import StyledTab from "./StyledTab";
import { TabProps } from "./types";
import { Text } from "../Text";

const Tab: React.FC<TabProps> = ({ isActive = false, onClick, children }) => (
  <StyledTab onClick={onClick} bgColor={isActive ? "textSubtle" : "input"} color={isActive ? "card" : "textSubtle"}>
    <Text fontWeight={600} color={isActive ? "card" : "textSubtle"} m={"0 8px"}>
      {children}
    </Text>
  </StyledTab>
);
export default Tab;
