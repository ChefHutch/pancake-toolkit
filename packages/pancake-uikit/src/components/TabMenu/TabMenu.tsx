import React, { cloneElement, Children, ReactElement } from "react";
import styled from "styled-components";
import Flex from "../Box/Flex";
import StyledTabMenu from "./StyledTabMenu";
import { TabMenuProps } from "./types";

const Wrapper = styled(Flex)`
  border-bottom: 2px solid ${({ theme }) => theme.colors.textSubtle};
  padding: 0 16px;
  justify-content: space-around;

  ${({ theme }) => theme.mediaQueries.sm} {
    justify-content: flex-start;
  }
`;

const ButtonMenu: React.FC<TabMenuProps> = ({ activeIndex = 0, onItemClick, children }) => {
  return (
    <Wrapper>
      <StyledTabMenu>
        {Children.map(children, (child: ReactElement, index) => {
          return cloneElement(child, {
            isActive: activeIndex === index,
            onClick: onItemClick ? () => onItemClick(index) : undefined,
          });
        })}
      </StyledTabMenu>
    </Wrapper>
  );
};

export default ButtonMenu;
