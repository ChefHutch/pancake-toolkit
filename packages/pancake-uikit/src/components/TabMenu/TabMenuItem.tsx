import React from "react";
import { ActiveTab, InactiveTab } from "./StyledTabs";
import { TabMenuItemProps } from "./types";

const TabMenuItem: React.FC<TabMenuItemProps> = ({ isActive = false, children }) => {
  if (!isActive) {
    return <InactiveTab>{children}</InactiveTab>;
  }

  return <ActiveTab>{children}</ActiveTab>;
};

export default TabMenuItem;
