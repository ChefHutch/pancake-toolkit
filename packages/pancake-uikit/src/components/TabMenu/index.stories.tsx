import React, { useState } from "react";
/* eslint-disable import/no-unresolved */
import { Meta } from "@storybook/react/types-6-0";
import TabMenu from "./TabMenu";
import TabMenuItem from "./TabMenuItem";

export default {
  title: "Components/Tab Menu",
  component: TabMenu,
  argTypes: {},
} as Meta;

export const Default: React.FC = () => {
  const [index, setIndex] = useState(0);
  const handleClick = (newIndex) => setIndex(newIndex);

  return (
    <div>
      <TabMenu activeIndex={index} onItemClick={handleClick}>
        <TabMenuItem>Total</TabMenuItem>
        <TabMenuItem>Cakers</TabMenuItem>
        <TabMenuItem>Flippers</TabMenuItem>
        <TabMenuItem>Storm</TabMenuItem>
      </TabMenu>
    </div>
  );
};
