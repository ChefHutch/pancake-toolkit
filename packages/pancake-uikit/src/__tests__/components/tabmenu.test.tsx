import React from "react";
import { renderWithTheme } from "../../testHelpers";
import { TabMenu, Tab } from "../../components/TabMenu";

const handleClick = jest.fn();

it("renders correctly", () => {
  const { asFragment } = renderWithTheme(
    <TabMenu activeIndex={0} onItemClick={handleClick}>
      <Tab>Item 1</Tab>
      <Tab>Item 2</Tab>
    </TabMenu>
  );
  expect(asFragment()).toMatchSnapshot();
});
