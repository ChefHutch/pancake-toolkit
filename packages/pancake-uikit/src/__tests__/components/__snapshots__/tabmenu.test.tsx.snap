// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ sc-gsTCUz sc-dlfnbm iwJkGQ ckYhbt gjsBuP"
  >
    <button
      class="sc-hKgILt flOdUK"
      color="card"
    >
      <div
        class="sc-eCssSg iJPpdt"
        color="card"
        font-weight="600"
      >
        Item 1
      </div>
    </button>
    <button
      class="sc-hKgILt giymoc"
      color="textSubtle"
    >
      <div
        class="sc-eCssSg gbqlTw"
        color="textSubtle"
        font-weight="600"
      >
        Item 2
      </div>
    </button>
  </div>
</DocumentFragment>
`;
